### PR TITLE
GH-3: Add ability to exclude all files in `.gitignore` from linting (resolves #3).

### LINKED ISSUE #3: Exclude files in .gitignore from linting
## Is your feature request related to a problem?

As per https://github.com/fluid-project/fluid-lint-all/pull/2#discussion_r558438411, fluid-lint-all currently lints generated files which are listed in a project's `.gitignore` file.

## Describe the solution you'd like

I think it's safe to assume that fluid-lint-all should ignore any files present in `.gitignore` by default. I think it's also good to ensure that:

- Explicitly naming a file in a linter's `includes` configuration block would cause it to be linted even if it was present in `.gitignore`
- The flag (which defaults to `true`) that excludes files in `.gitignore` could be passed with a value of false to remove the automatic exclusion.

## Describe alternative solutions you've considered

Not applicable.

## Additional context or notes

Not applicable.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See #3 for details.

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
@greatislander, I cut `1.0.5-dev.20210201T163615Z.cff0a11.GH-3` for you to try out in a project that uses a `.gitignore`.

**Comment by greatislander** [comment]
Works great, @the-t-in-rtf — I updated Trivet to use that dev release and I was able to remove my linting exclusions for ignored files (https://github.com/fluid-project/trivet/commit/54122d27fb1d9dee70208c57be2d695c82cc9f6f).

File: tests/js/launcher-tests.js [hunk 1]
```diff
@@ -129,7 +129,18 @@ fluid.defaults("fluid.tests.lintAll.launcher.runner", {
             configFile: ".fluidlintallrc-nofiles.json",
             shouldBeInvalid: true,
             expectedMessage: ["ERROR: No files checked, please review your configuration and command line arguments."]
+        },
+        useGitIgnore: {
+            message: "We should be able to exclude files using a .gitignore file.",
+            configFile: ".fluidlintallrc-gitignore.json"
+        },
+        disableGitIgnore: {
+            message: "We should be able to exclude files using a .gitignore file.",
```
Comment (the-t-in-rtf) [inline tests/js/launcher-tests.js:138]: This needs to be reworked, it's obvious copypasta as well.

Comment (the-t-in-rtf) [reply tests/js/launcher-tests.js]: Fixed.

**Pull request merged**